### DISCUSSION TITLE
Select returns less rows than expected

### DISCUSSION THREAD

**Discussion by dsimog01** [prisma/prisma#16918] [Q&A] [answered]
I Prisma 4.7.1 I am performing a findMany operation that should return 43 rows, but I am getting only 36 and I cannot figure out why. 

For the same DB, I execute the following command through MySQL that returns 43 rows:
```
SELECT products FROM OrderedProducts WHERE address_status="NOT_WHITELISTED" AND order_number IN (SELECT order_number FROM Orders WHERE whitelist_status="APPROVED");

43 rows in set (0.00 sec)
```

However, the function getApprovedAddresses() of the following TS code for Prisma returns only 36:
```
public async getOrdersID(where: Prisma.OrdersWhereInput) {
  const orders = await this.prisma.orders.findMany({
    select: {
      order_number: true,
    },
    where: where,
  });

  return orders.map((order) => order.order_number);
}


public async getApprovedAddresses(
): Promise<AddressInfo[]> {
  const approvedOrders = await this.getOrdersID({
    whitelist_status: WhitelistStatus.APPROVED,
  });

  return await this.prisma.orderedProducts.findMany({
    select: {
      eth_address: true,
    },
    where: {
      order_number: {
        in: approvedOrders,
      },
      address_status: AddressStatus.NOT_WHITELISTED,
    },
  });

  return addressInfo;
}
```

Any help is appreciated :pray:

**Comment by dsimog01** [comment, marked as answer]
It was a problem with the threads... a foreach with awaits instead of for. This discussion can be closed.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
